buildSrc/src/main/kotlin/org/gradle/plugins/compile/AvailableJavaInstallations.kt

```diff
@@ -48,8 +70,52 @@ open class AvailableJavaInstallations(javaHomesForCompilation: List<String>, jav
         else -> throw IllegalArgumentException("No Java installation found which supports Java version $javaVersion")
     }
 
-    val validBuildCacheConfiguration
-        get() = !jdkForCompilation(JavaVersion.VERSION_1_7).current && jdkForCompilation(JavaVersion.VERSION_1_8).current
+    fun validateBuildCacheConfiguration(buildCacheConfiguration: BuildCacheConfiguration) {
+        val validationErrors = validateCompilationJdks()
+        if (validationErrors.isNotEmpty()) {
+            val message = formatValidationError(
+                "In order to have cache hits from the remote build cache, your environment needs to be configured accordingly!",
```

Comment:
Can we be more helpful here, like say what exactly needs to be configured?

We could mention how to set up the remote cache in the `REAMDE`, too, maybe?

Reply:
We print the validation errors (i.e. what is missing) right below that message, e.g.:
```
* What went wrong:
A problem occurred configuring root project 'gradle'.
> In order to have cache hits from the remote build cache, your environment needs to be configured accordingly!
  Problems found:
      - Must set project or system property 'java7Home' to the path of an Oracle JDK 7, is currently unset.
```
or
```
* What went wrong:
A problem occurred configuring root project 'gradle'.
> In order to have cache hits from the remote build cache, your environment needs to be configured accordingly!
  Problems found:
      - Must set project or system property 'java7Home' to the path of an Oracle JDK 7, is currently Oracle JDK 8 at .../java/8u151-oracle.
```

Isn't that enough? What would additional documentation provide?

Reply:
Sorry, I missed that the details were in the details.